### ISSUE TITLE
Output file is not generated if manifest is invalid

### ISSUE THREAD

**Issue by MariamKhalatova** [Green-Software-Foundation/if#701]
## Description of the Error

Output file with error is not saved if manifest structure is invalid.

## Expected Behaviour

If manifest structure is invalid, output file should be saved and the validation error should be present in the execution.error.

## Actual Behaviour

Error is shown in console, no output file is saved.

## Steps to Reproduce

1. Run the manifest provided below.

## Manifest File That Generated the Error

```yaml
name: basic
description: a minimal manifest executing a single plugin on a single component for a single timestep
tags:
initialize:
  outputs: ['yaml']
  plugins:
    teads-curve: 
      path: '@grnsft/if-unofficial-plugins'
      # method: TeadsCurve
      global-config:
        interpolation: spline
tree:
  children:
    child-0:
      defaults:
        cpu/thermal-design-power: 100
      pipeline:
        - teads-curve
      inputs:
        - timestamp: 2023-07-06T00:00
          duration: 1
          cpu/utilization: 70
        - timestamp: 2023-07-06T00:01
          duration: 1
          cpu/utilization: 80
        - timestamp: 2023-07-06T00:02
          duration: 1
          cpu/utilization: 20

```

Command:
```
npm run ie -- --manifest manifests/examples/basic.yml --output manifests/outputs/basic
```

## Links to Any Additional Code

<img width="1246" alt="Screenshot 2024-05-07 at 6 28 06 PM" src="https://github.com/Green-Software-Foundation/if/assets/109615222/28ab1986-3ded-4c94-b031-431ac573e778">

## Runtime Info

macos, node 18

**Comment by zanete** [comment]
Will be fixed by #592

**Comment by MariamKhalatova** [comment]
fixed by #691
